test/powershell/Language/Operators/SplitOperator.Tests.ps1

```diff
@@ -0,0 +1,236 @@
+            $res.count | Should Be 2
+            $res[0] | Should Be "a12a`n"
+            $res[1] | Should Be "34b`nc56c`nd78d"
+        }
+
+        It "Binary split operator works with Singleline,Multiline" {
+            $res = $testText -split 'b.+c', 0, 'Singleline,Multiline'
+            $res.count | Should Be 2
+            $res[0] | Should Be "a12a`n"
+            $res[1] | Should Be "`nd78d"
+
+            $res = $testText2 -split 'b.+c', 0, 'Singleline,Multiline'
+            $res.count | Should Be 2
+            $res[0] | Should Be "a12a`n#"
+            $res[1] | Should Be "`nd78d"
+
+            $res = $testText -split '^b.+c', 0, 'Singleline,Multiline'
+            $res.count | Should Be 2
+            $res[0] | Should Be "a12a`n"
+            $res[1] | Should Be "`nd78d"
+
+            $res = $testText2 -split '^b.+c', 0, 'Singleline,Multiline'
+            $res.count | Should Be 1
+        }
+
+        It "Binary split operator works with IgnorePatternWhitespace" {
+            $res = "a: b:c" -split ': '
+            $res.count | Should Be 2
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b:c"
+
+            $res = "a: b:c" -split ': ',0,'IgnorePatternWhitespace'
+            $res.count | Should Be 3
+            $res[0] | Should Be "a"
+            $res[1] | Should Be " b"
+            $res[2] | Should Be "c"
+        }
+
+        It "Binary split operator works with ExplicitCapture" {
+            $res = "a:b" -split "(:)"
+            $res.count | Should Be 3
+            $res[0] | Should Be "a"
+            $res[1] | Should Be ":"
+            $res[2] | Should Be "b"
+
+            $res = "a:b" -split "(:)", 0, 'ExplicitCapture'
+            $res.count | Should Be 2
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b"
+        }
+
+        It "Binary split operator works with SimpleMatch" {
+            $res = "abc" -split "B", 0, 'SimpleMatch,IgnoreCase'
+            $res.count | Should Be 2
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "c"
+        }
+
+        It "Binary split operator works with RegexMatch" {
+            $res = "abc" -split "B", 0, 'RegexMatch,Singleline'
+            $res.count | Should Be 2
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "c"
+        }
+
+        It "Binary split operator doesn't works with RegexMatch,SimpleMatch" {
+            { "abc" -split "B", 0, 'RegexMatch,SimpleMatch' } | ShouldBeErrorId "InvalidSplitOptionCombination"
+        }
+    }
+
+    Context "Unary split operator" {
+        It "Unary split operator has higher precedence than a comma" {
+            $res = -split "a b", "c d"
+            $res.count | Should Be 2
+            $res[0][0] | Should Be "a"
+            $res[0][1] | Should Be "b"
+            $res[1] | Should Be "c d"
+        }
+
+        It "Unary split operator can split array of values" {
+            $res = -split ("a b", "c d")
```

Comment:
wow didn't know you can do that!

Reply:
@vors: Regarding the _array_-as-operand aspect (not the unary-operator aspect):

Since the only operand unary `-split` accepts is of type `[string]`, the array is simply stringified.

In other words, assuming the default `$OFS` value:

       -split ("a b", "c d")

is the same as:

       -split "a b c d"

What puzzles me, however is that the implicit array-to-string conversion in this case does NOT respect `$OFS`:

```powershell
       > & { $OFS='@'; $a = 'a b', 'c d'; (-split $a).Count; (-split "$a").Count }
       4 # !! $OFS not respected.
       3  # OK: the equivalent of (-split 'a b@c d').Count
```

Is this by design?

Reply:
@vors: 
Turns out I didn't know either, mostly because [the documentation](https://github.com/PowerShell/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Split.md):

*  in part suggests that only string _scalars_ are supported (`<String>` in the syntax diagram) 

* where it does mention arrays (multiple input strings), mistakenly claims that only the _binary_ form supports them ("To split more than one string, use the binary split operator ").

I've created a [doc issue](https://github.com/PowerShell/PowerShell-Docs/issues/1640) to address that.

(I originally thought that when you pass an array it simply gets stringified based on `$OFS`, but that's not the case.)